### PR TITLE
[Basic] Hold lane release pipe open until process termination

### PR DESCRIPTION
**Pull request by dmbryson**
Clients (especially bash on Linux) may receive SIGPIPE when we
close the lane release pipe immediately. Although we could require
clients to implement their own SIGPIPE traps, it is simpler to hold the
pipe open until we are sure the client is done.

rdar://problem/45196629

### CODE REVIEW

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
Yeah, this makes sense to me.

File: lib/Basic/Subprocess.cpp [hunk 1]
```diff
@@ -256,6 +261,7 @@ static void cleanUpExecutedProcess(
   ProcessStatus processStatus = cancelled ? ProcessStatus::Cancelled : (exitCode == 0) ? ProcessStatus::Succeeded : ProcessStatus::Failed;
   ProcessResult processResult(processStatus, exitCode, pid, utime, stime,
                               usage.ru_maxrss);
+  std::cout << "process finished: " << std::dec << exitCode << std::endl;
```
Comment (ddunbar) [inline lib/Basic/Subprocess.cpp]: Stray debug log?

File: lib/Basic/Subprocess.cpp [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #include "llvm/Support/Program.h"
 
 #include <atomic>
+#include <iostream>
```
Comment (ddunbar) [inline lib/Basic/Subprocess.cpp]: Stray include (I despise iostream, it adds tons of bloat).

File: lib/Basic/Subprocess.cpp [hunk 3]
```diff
@@ -528,19 +534,17 @@ void llbuild::basic::spawnProcess(
     }
 
     if (control.shouldRelease()) {
-      // Close the read end of the release pipe.
-      ::close(releasePipe[0]);
-
       releaseFn([
```
Comment (ddunbar) [inline lib/Basic/Subprocess.cpp]: Might be worth either putting a comment here, or at the other close site, explaining why we close when we do.

File: lib/Basic/Subprocess.cpp (hunk 2 above)
Comment (dmbryson) [reply lib/Basic/Subprocess.cpp]: definitely, that was also stray debug

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Pull request merged**